pkg/downwardmetrics/virtio-serial/server.go

```diff
@@ -181,6 +184,8 @@ func (s *downwardMetricsServer) serve(ctx context.Context, conn net.Conn) {
 			}
 		case <-ctx.Done():
 			return
+		case <-signalStopChan:
```

Comment:
The `case <-ctx.Done():` can never happen as the context is only canceled when serve returns, which it only can if the for loop is broken.